cccl

```diff
@@ -301,6 +319,77 @@ processargs $CCCL_OPTIONS
 IFS=""
 processargs $@
 
+# Process "print-search-dirs" and exit gracefully
+if $print_search_dirs; then
+    # Record the original "IFS"
+    if ! [ -z "$IFS" ]; then
+        originalIFS="$IFS"
+    fi
+
+    # Set the "winpath" binary to convert Windows paths to Unix paths.
+    winpath_bin="$CCCL_WINPATH_BIN" # "CCCL_WINPATH_BIN" must have the same format, e.g., as "cygpath"
+    if [ -z "$winpath_bin" ]; then
+        # Guess the "winpath" binary
+        IFS=" "
+        guess_winpath_bins_str="cygpath wslpath winepath" # For Cygwin/MSYS2 and native Windows, WSL and native Windows, and Linux and Wine, respectfully
+        read -ra guess_winpath_bins <<< "$guess_winpath_bins_str"
+        for guess_winpath_bin in ${guess_winpath_bins[@]}; do
+            winpath_bin="$(which $guess_winpath_bin)"
+            if ! [ -z "$winpath_bin" ]; then
+                break
+            fi
+        done
+        if [ -z "$winpath_bin" ]; then
+            echo "Cannot guess \"winpath\" binary to convert Windows paths to Unix paths. You must set \"CCCL_WINPATH_BIN\" with a format similar to \"cygpath\", \"wslpath\", or \"winepath\"."
+            exit 1
+        fi
+    fi
+
+    # Print "install"
+    echo "install: $script_dir"
+
+    # TODO: Implement printing "programs" for the MSVC toolchain. Not sure which bin dirs to print, nor if it is ever needed for
+    # the MSVC toolchain
+
+    # Set the global libraries
+    global_libraries=""
+    IFS=";"
+    read -ra global_libraries_windows <<< "$LIB"
+    for libpath in ${global_libraries_windows[@]}; do
+        # Using "realpath" is undesirable, but it might be the only solution to have a path without ":" when the
+        # the Windows drive letter is present in the path. For instance, a path with "c:" in Wine will be converted
+        # to a path with "drive_c". The reason for avoiding paths with ":" is that ":" is generally unescapable in Unix
+        # (e.g. in "PATH" environment variable).
+        unix_global_library_path="$(realpath $($winpath_bin -u $libpath))"
```

Comment:
I don't like to use `realpath` here, but it is the only way I can think of to remove the `:` from the Wine drives, for e.g.

I believe this argument is mostly necessary for `libtool` to search for libraries when creating dynamic libraries, and the issue when going to the parent folder (i.e. `../`) won't happen, since only the directories in `libraries` will be searched.

Reply:
I resolved this issue by shortening only the Unix portion of the Windows path prefix, by parsing out the prefix using `Python`'s `pathlib`'s `PureWindowsPath`.